docs/config_entries_diagnostics.md

```diff
@@ -0,0 +1,40 @@
+---
+title: Config entry diagnostics
+---
+
+Config entries can provide diagnostics to help the user gather data about an integration and do troubleshooting. Diagnostics can be provided for the config entry but also individually for each device entry.
```

Comment:
Let's also explain where they can find this, I am not sure how many pictures we usually do, but it would be nice to give a complete picture

Reply:
I've added a short explanation. I am not sure about adding screenshots, as they may get outdated. Let me know if you think the explanation is not enough and we should still add screenshots.